### ISSUE TITLE
setmeta cannot change custom meta with uppercase chars

### ISSUE THREAD

**Issue by xmedeko** [GoogleCloudPlatform/gsutil#421]
I have inserted an object with custom metadata `X-Ice:1` by the Python `google-api-python-client` and I see it in the web console. I want to remove it by the:
```
gsutil setmeta -h 'x-goog-meta-X-Ice' 'gs://bucket/file'
```
but it does not remove the metadata. In the web console I see a new metadata `x-ice` instead. When i try the lowercase:
```
gsutil setmeta -h 'x-goog-meta-x-ice' 'gs://bucket/file'
```
then the `x-ice` is removed, but the `X-Ice` remains.

I think the `gsutil`, `google-api-python-client` and the Storage web console should have same policy: either preserve casing or make the custom metadata lowercase.

----
gsutil version: 4.20

**Comment by houglum** [comment]
I can't reproduce this on newer versions - I tested with gsutil 4.24 (the newest version is 4.25).  Can you try with a current version?

**Comment by mhoug89** [comment]
Yep, looks like it was fixed with the 4.23 release with via commit:
https://github.com/GoogleCloudPlatform/gsutil/commit/254959b1f1fbada9df6bd5c423c63b5ac214af1d

**Comment by xmedeko** [comment]
You are right, it works in 4.24. I apologize for not trying the newest version first.
